doc/api/next_api_changes/development/30767-MS.rst

```diff
@@ -0,0 +1,18 @@
+``legend.linewidth`` rcParam and parameter
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A new rcParam ``legend.linewidth`` has been added to control the line width of
+the legend's box edges. When set to ``None`` (the default), it inherits the
+value from ``axes.linewidth``. This allows for independent control of the
+legend frame line width without affecting other elements.
+
+.. versionadded:: 3.10
+
+The `.Legend` constructor also accepts a new *linewidth* parameter to set the
+legend frame line width directly, overriding the rcParam value::
+
```

Comment:
Using the `plot` directive here will mean we get an output image with the example.  See [here](https://github.com/matplotlib/matplotlib/blob/dedfe9be48ad82cade86766ef89410844ff09b31/doc/release/next_whats_new/separated_hatchcolor.rst?plain=1#L11-L13) for an example of how that’s done.